Core/CRFSharp/encoder/ModelWriter.cs

```diff
@@ -5,24 +5,32 @@
 using System.Text;
 using AdvUtils;
 using System.Threading.Tasks;
+using CRFSharp.decoder;
 
 namespace CRFSharp
 {
-    public class ModelWritter : BaseModel
+    public class ModelWriter : BaseModel
     {
+        private readonly string modelFileName;
+
         private readonly Pool<StringBuilder> _buildersPool =
             new Pool<StringBuilder>(p => new StringBuilder(100), b => b.Clear());
 
+        private readonly ModelReader modelReader;
```

Comment:
ModelReader was historically used in ModelWriter class to read the model for retraining. I just moved it into a class field. Here it is in the original code: https://gist.github.com/amironoff/82ff62db7c9fb112979ff442b4112d1d

On the other hand, you're right, ModelReader isn't always needed here, just when retraining. Therefore, I moved it back into the training method.